### DISCUSSION TITLE
How to set icons for specific file types like in icons.example

### DISCUSSION THREAD

**Discussion by psi4j** [gokcehan/lf#1024] [Q&A] [answered]
I copied the contents of icons.example into my own icons file in the `~/.config/lf` directory, next to my lfrc, and it is showing all of the standard file types (listed below), but not all of the other types, such as file extensions with the *wildcard. Does anybody know how to configure these? 
```ln		# LINK
or		# ORPHAN
tw	t	# STICKY_OTHER_WRITABLE
ow		# OTHER_WRITABLE
st	t	# STICKY
di	ﱮ	# DIR
pi	p	# FIFO
so	s	# SOCK
bd	b	# BLK
cd	c	# CHR
su	u	# SETUID
sg	g	# SETGID
ex		# EXEC
fi		# FILE
```

**Comment by janos-r** [comment]
Let me try to rephrase your question, because I think Im searching for the same solution:
There is an inconsistency in the icon selection!
For example: if a file is type `fi`, the icon for the file extension is picked.
But when a file is type `ex`, the ex icon gets picked, instead of matching for the file extension icon bellow.

**Reply by psi4j** [reply]
Precisely. This is what I'm experiencing. LSD does something similar to LF and has some sort of method to differentiate the various file extensions. LSD is written in Python though. I'm unfamiliar with Go, or I'd try to figure it out myself. Regardless of the icons, LF is the fastest way I've found to engage with your file system through the terminal. I'm exited to see where its development goes over the coming years.

**Comment by janos-r** [comment]
Has probably nothing to do with the language, its probably simply just a bug.
Im annoyed how every single "executable", which are almost all my old files from windows and others, have a rocket, instead of a pdf for example.
I hope somebody takes notice and can fix it.

**Reply by CatsDeservePets** [reply]
There is still an open issue regarding this https://github.com/gokcehan/lf/issues/992.

**Comment by CatsDeservePets** [comment, marked as answer]
Wildcards work like this
*.py 🐍

Regarding the executable icon, there is sadly still no solution that  makes everyone happy: https://github.com/gokcehan/lf/issues/992
